### PR TITLE
Cleanup Ingest Budget v1 related deprecated resources

### PR DESCRIPTION
**Pull request by vsinghal13**
In this PR, we are cleaning up the Ingest Budget V1 related resources, which are already deprecated.
 - `resource_sumologic_ingest_budget`
 - `resource_sumologic_collector_ingest_budget_assignment`
This is a `breaking change`, so it will be released as a major version release.

### CODE REVIEW

**Comment by vsinghal13** [comment]
Soft removal was done in https://github.com/SumoLogic/terraform-provider-sumologic/pull/684

File: CHANGELOG.md
Comment (ashpaliwal) [inline CHANGELOG.md:6]: Can you merge this section with unreleased, removing version number keeping rest content same

**Review by ashpaliwal — APPROVED** [review]
LGTM, Please merge once your team approves the changes

**Review by nchudasmasumo — APPROVED** [review]

**Pull request merged**
